### PR TITLE
[6.x] Use Dialog for Article links, when "modal option" is selected

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes
Use Dialog for Artcile links, when "modal option" is selected.

**And new feature**: Allow to show inline content in the popup.

### Testing Instructions

Add a couple links to the Article,
First link: any regular URL.
Second link: `#my-popup-target`.
Also add to the article content, following html:
```html
<div class="hidden">
<div id="my-popup-target">
<p class="p-3">Lorem ipsum dolor sit amet, cu vis velit signiferumque, eos no possim mollis facilisi.</p>
</div>
</div>
```

For both links select option: show in modal.

Got to article view, and click these links.

### Actual result BEFORE applying this Pull Request
First link works in BS Modal:
Screenshot 2023-12-03_17-59-26
Second link does nothing.

### Expected result AFTER applying this Pull Request
First link works with Joomla Dialog:
Screenshot 2023-12-04_11-07-49

Second link works with Joomla Dialog:
Screenshot 2023-12-04_11-21-21

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: TBD
- [ ] No documentation changes for manual.joomla.org needed

References:

- https://github.com/joomla/joomla-cms/pull/40150

### CODE REVIEW

**Comment by fgsw** [comment]
Test successfully but can't mark it on issues.joomla.org, get ```Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 8.1.0".```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7801a6a05aaab37c726598f547ef047aeedb5eac<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42461">issues.joomla.org/tracker/joomla-cms/42461</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42461">issues.joomla.org/tracker/joomla-cms/42461</a>.</sub>

**Comment by LadySolveig** [comment]
Thank you @Fedik for this nice work!

A direct change of the modal option for the links in article could lead to broken styles and functions in the frontend for some users in conjunction with a Cassiopeia child template, for example.
If a template uses the bootstrap modal events with custom javascript to perform another action when opening or closing, this will unexpectedly stop working. Similarly, custom styles for the modals have to be rewritten from the users and custom css for this purpose will unexpectedly not work anymore.

Perhaps another select option for the new dialog would be a variant to prevent this. 
And the deprecation of the BS modal in the frontend for the article links could be the long-term solution.

What do you think about this?
Would be glad to hear further opinions as well.

**Comment by Fedik** [comment]
I can do another option, it is not a problem for me.
But I have seen that current modal option looks bad anyway.

What other thinks? :)

**Comment by Fedik** [comment]
On second thought, another "modal" option will be very confusing, we can probably close it and keep BS here, or rebase to 6-dev

**Comment by LadySolveig** [comment]
> On second thought, another "modal" option will be very confusing

That's exactly what bothers me a little. But I think the new dialogue looks much better by default even without adjusting the css. Tricky :)  I will try if I can get a few more opinions on this.

**Comment by LadySolveig** [comment]
@Fedik thank you for your work. We decided to rebase to 6

File: components/com_content/tmpl/article/default_links.php [hunk 1]
```diff
@@ -10,7 +10,8 @@
 
 defined('_JEXEC') or die;
 
-use Joomla\CMS\HTML\HTMLHelper;
+/** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
+$wa = $this->document->getWebAssetManager();
```
Comment (laoneo) [inline components/com_content/tmpl/article/default_links.php]: Can you use here $this->getDocument()?

Comment (Fedik) [reply components/com_content/tmpl/article/default_links.php]: Not like `$this->document` was "official" in component `tmpl`? :)
But sure, can be anything, I will update it when time of 6.x will come

Comment (laoneo) [reply components/com_content/tmpl/article/default_links.php]: Thanks, will put this one then on hold till updated.

**Comment by Hackwar** [comment]
Can you please make the requested change, so that we can test this during PBF? Then we can already merge this into 6.0.

**Comment by SarahBerryman12** [comment]
I have tested this item :white_check_mark: successfully on b1872fffd9f20733fcafdb2146154aa2c6f9d0bc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42461">issues.joomla.org/tracker/joomla-cms/42461</a>.</sub>

**Comment by eddiekonczal** [comment]
I have tested this item :white_check_mark: successfully on b1872fffd9f20733fcafdb2146154aa2c6f9d0bc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42461">issues.joomla.org/tracker/joomla-cms/42461</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42461">issues.joomla.org/tracker/joomla-cms/42461</a>.</sub>

**Comment by brianteeman** [comment]
How can this have been tested with 6.x

**Comment by laoneo** [comment]
@brianteeman the pr was made against the 6.0-dev branch. This is the branch where we commit all the changes which should go into 6.0.

**Comment by ceford** [comment]
I updated the Help page at https://docs.joomla.org/Help5.x:Articles:_Edit to indicate the new feature in J6.

**Pull request merged**